plugins/chat/assets/javascripts/discourse/models/chat-channel.js

```diff
@@ -133,15 +133,35 @@ export default class ChatChannel {
   }
 
   get unreadThreadCount() {
-    return this.unreadThreadIds.size;
+    return this.unreadThreadOverview.size;
   }
 
-  get unreadThreadIds() {
-    return this._unreadThreadIds;
+  get unreadThreadOverview() {
+    return this._unreadThreadOverview;
   }
 
-  set unreadThreadIds(unreadThreadIds) {
-    this._unreadThreadIds = new TrackedSet(unreadThreadIds);
+  set unreadThreadOverview(unreadThreadOverview) {
+    this._unreadThreadOverview = new TrackedMap();
```

Comment:
Why do we re-init it here?

Reply:
Because we're setting the whole thing based on fresh data from the API, I thought it would be easier than comparing what is there vs what is not. We did the same thing with TrackedSet previously.

Reply:
I think you have `clear()` on map

Reply:
Yep we do, will change to use that :+1: